### PR TITLE
Challenge 07: POST request, built extensive error handling 

### PR DESCRIPTION
**Pull request by messat**
1. Model has custom error handling depending on the status - error code
2. Altered the error handling block. 
3. Sent a body to add comments to the comments table 
4. Added express middleware to extract the body

### CODE REVIEW

**Comment by messat** [comment]
Merged 07 challenge with main branch

File: models/api.model.js [hunk 1]
```diff
@@ -42,4 +44,33 @@ exports.selectCommentsByArticleId =  (article_id)=>{
             return Promise.reject({status: 404, msg: '404 Not Found'})
         }
     })
+}
+
+exports.insertNewComment= (article_id, newComment)=>{
+     return db.query('SELECT * FROM comments WHERE article_id = $1', [article_id])
```
Comment (MRombi) [inline models/api.model.js:50]: We have a model that selects articles by article id already, we can reutilise it 🙂

File: __tests__/newsApi.test.js [hunk 2]
```diff
@@ -194,4 +194,67 @@ describe('CORE Task 6: GET request /api/articles/:article_id/comments', () => {
...
+            expect(res.body.comment.author).toBe('butter_bridge')
+            expect(res.body.comment.body).toBe('I wake up in the morning thinking why do I even bother')
+            expect(res.body.comment).toEqual({
+              comment_id: 19,
+              body: 'I wake up in the morning thinking why do I even bother',
+              article_id: 1,
+              author: 'butter_bridge',
+              votes: 0,
+              created_at: expect.any(String)
+            })
```
Comment (MRombi) [inline __tests__/newsApi.test.js:219]: We could add an extra test to check what happens if we have some additional properties, which should send a 201 status and ignore the extra properties 🙂

**Review by MRombi — COMMENTED** [review]
Well done so far, check my feedback for next steps 🙂

**Pull request merged**